### ISSUE TITLE
Bug: MA broken on restart : Non-thread-safe operation invoked - 3.1.0-beta

### ISSUE THREAD

**Issue by Julien-Decoen** [jseidl/hass-magic_areas#313]
**Describe the bug**
When I reboot HA, MA throw this error and entities become unavailable

<img width="1038" alt="image" src="https://github.com/jseidl/hass-magic_areas/assets/22341879/9497843e-973d-4c17-900d-79f93acb1f22">

<img width="1363" alt="image" src="https://github.com/jseidl/hass-magic_areas/assets/22341879/04c4d932-787c-4530-a3b2-32c46f73a222">

**To Reproduce**
1. fresh install of home assistant by removing .storage directory and the database
2. Running HA with vscode - Home Assistant (debug)
3. Add a MA device with config flow. It's working
4. Restart HA then device is unavailable

**Screenshots**

**Additional context**

Logs : 

```2024-02-08 18:57:10.868 ERROR (SyncWorker_3) [custom_components.magic_areas.util] [Cuisine] Error loading platform entities on <function add_sensors at 0xffff4c812700>.
Traceback (most recent call last):
  File "/workspaces/linus-assistant-config/config/custom_components/magic_areas/util.py", line 67, in load_entities
    callback_fn(area, async_add_entities)
  File "/workspaces/linus-assistant-config/config/custom_components/magic_areas/binary_sensor.py", line 77, in add_sensors
    async_add_entities([AreaPresenceBinarySensor(area)])
  File "/home/vscode/.local/lib/python3.11/site-packages/homeassistant/helpers/entity_platform.py", line 482, in _async_schedule_add_entities_for_entry
    task = self.config_entry.async_create_task(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/homeassistant/config_entries.py", line 859, in async_create_task
    task = hass.async_create_task(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/homeassistant/core.py", line 633, in async_create_task
    task = self.loop.create_task(target, name=name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 436, in create_task
    task = tasks.Task(coro, loop=self, name=name, context=context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 763, in call_soon
    self._check_thread()
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 800, in _check_thread
    raise RuntimeError(
RuntimeError: Non-thread-safe operation invoked on an event loop other than the current one
2024-02-08 18:57:10.871 ERROR (SyncWorker_3) [homeassistant] Error doing job: Task was destroyed but it is pending!:   File "/home/vscode/.vscode-server/extensions/ms-python.python-2024.0.1/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydev_bundle/pydev_monkey.py", line 1118, in __call__
    ret = self.original_func(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.11/threading.py", line 995, in _bootstrap
    self._bootstrap_inner()
  File "/usr/local/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 83, in _worker
    work_item.run()
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/workspaces/linus-assistant-config/config/custom_components/magic_areas/util.py", line 67, in load_entities
    callback_fn(area, async_add_entities)
  File "/workspaces/linus-assistant-config/config/custom_components/magic_areas/binary_sensor.py", line 77, in add_sensors
    async_add_entities([AreaPresenceBinarySensor(area)])
  File "/home/vscode/.local/lib/python3.11/site-packages/homeassistant/helpers/entity_platform.py", line 482, in _async_schedule_add_entities_for_entry
    task = self.config_entry.async_create_task(
  File "/home/vscode/.local/lib/python3.11/site-packages/homeassistant/config_entries.py", line 859, in async_create_task
    task = hass.async_create_task(
  File "/home/vscode/.local/lib/python3.11/site-packages/homeassistant/core.py", line 633, in async_create_task
    task = self.loop.create_task(target, name=name)
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 436, in create_task
    task = tasks.Task(coro, loop=self, name=name, context=context)

2024-02-08 18:57:10.871 WARNING (SyncWorker_3) [py.warnings] /workspaces/linus-assistant-config/config/custom_components/magic_areas/util.py:69: RuntimeWarning: coroutine 'EntityPlatform.async_add_entities' was never awaited
  _LOGGER.exception(f"[{area.name}] Error loading platform entities on {str(callback_fn)}.")

**Comment by Julien-Decoen** [comment]
I forgot to mention that I make a fresh install of home assistant by removing .storage directory and the database.

**Comment by jseidl** [comment]
This might be related to either VS code environment or a fresh install, I'll setup a fresh environment and try it out.

**Comment by jseidl** [comment]
Closed by #315. Thanks @Julien-Decoen!
